src/pathfinding_service/model/token_network.py

```diff
@@ -198,8 +183,51 @@ def is_valid(self) -> bool:
                 )
                 return False
 
+        # check capacity with fees
+        valid_fees = self._calculate_fees()
+        if not valid_fees:
+            return False
+
+        fees = self.fees + [FeeAmount(0)]  # The hop to the target does not incur mediation fees
+        for edge, fee in zip(edges, fees):
+            # check capacity
+            if edge["view"].capacity < required_capacity:
+                log.debug(
+                    "Path invalid because of missing capacity (with fees)",
+                    edge=edge,
+                    fee=fees,
+                    available_capacity=edge["view"].capacity,
+                    required_capacity=required_capacity,
+                )
+                return False
+            required_capacity = PaymentAmount(required_capacity + fee)
+
         return True
 
+    @property
+    def edge_attrs(self) -> Iterable[dict]:
+        return (self.G[node1][node2] for node1, node2 in zip(self.nodes[:-1], self.nodes[1:]))
+
+    def to_dict(self) -> dict:
+        return dict(
+            path=[to_checksum_address(node) for node in self.nodes], estimated_fee=sum(self.fees),
+        )
+
+    @property
+    def is_valid(self) -> bool:
```

Comment:
We don't need this as a property, anymore. We can just rename `_is_valid` to `is_valid`.